File: frontend/lib/api/ai.ts
import {
  authenticatedFetch,
  getApiError,
} from "@/lib/api/client";

interface GenerateProductDescriptionInput {
  name: string;
  category: string;
  is_customizable: boolean;
}

interface ProductDescriptionResponse {
  description: string;
}

interface BusinessAssistantResponse {
  answer: string;
}

export async function generateProductDescription(
  input: GenerateProductDescriptionInput,
): Promise<string> {
  const response = await authenticatedFetch(
    "/api/ai/product-description",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(input),
    },
  );

  if (!response.ok) {
    throw new Error(await getApiError(response));
  }

  const data =
    (await response.json()) as ProductDescriptionResponse;

  return data.description;
}

export async function askBusinessAssistant(
  question: string,
): Promise<string> {
  const response = await authenticatedFetch(
    "/api/ai/business-assistant",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        question: question.trim(),
      }),
    },
  );

  if (!response.ok) {
    throw new Error(await getApiError(response));
  }

  const data =
    (await response.json()) as BusinessAssistantResponse;

  return data.answer;
}

File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field


class ProductDescriptionRequest(BaseModel):
    name: str = Field(min_length=2, max_length=150)
    category: str = Field(min_length=2, max_length=100)
    is_customizable: bool = False


class ProductDescriptionResponse(BaseModel):
    description: str

class BusinessAssistantRequest(BaseModel):
    question: str = Field(
        min_length=3,
        max_length=500,
    )


class BusinessAssistantResponse(BaseModel):
    answer: str